instrumentation/nocode/src/test/java/com/splunk/opentelemetry/instrumentation/nocode/JSPSTest.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.instrumentation.nocode;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
+public class JSPSTest {
+  private static final Map<String, String> thiz = new HashMap<>();
+  private static final Set<String> param0 = new HashSet<>();
+
+  static {
+    thiz.put("key", "value");
+    param0.add("present");
+  }
+
+  @Test
+  public void testBasicBehavior() {
+    String[][] tests =
+        new String[][] {
+          // Might be nice to support a bare "param0" or "this" but as a workaround can always use
+          // "this.toString()"
+          {"this.toString()", "{key=value}"},
+          {"this.toString().length()", "11"},
+          {"this.get(\"key\")", "value"},
+          {"this.get(\"key\").substring(1)", "alue"},
+          {"param0.isEmpty()", "false"},
+          {"param0.contains(\"present\")", "true"},
+          {"this.entrySet().size()", "1"},
+        };
+    for (String[] test : tests) {
+      assertEquals(test[1], JSPS.evaluate(test[0], thiz, new Object[] {param0}), test[0]);
+    }
+  }
```

Comment:
Use a proper [parameterized tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests) please! Also applies to methods below.

Reply:
TIL, thanks!  My style was JUnit circa 1999.